### DISCUSSION TITLE
Downloading files using pb.send()

### DISCUSSION THREAD

**Discussion by mukhtharcm** [pocketbase/pocketbase#5506] [Q&A]
I'm trying to create an endpoint on my custom pocketbase that creates a pdf report using and fpdf and give back.

I created the endpoint, when going directly to the URL, I'm able to download the file.

Also, when I use fetch like this,
```ts
  
			let pbBaseUrl = pb.baseUrl;

			let fileUrl = pbBaseUrl + endpoint;

			const response = await fetch(fileUrl, {
				method: 'GET',
				headers: {
					Accept: 'application/pdf',
					Authorization: pb.authStore.token
				}
			});
```
I'm able to download the file.

But, when I try to do the same with pb.send() the file is not properly getting downloaded

```ts
const response = await pb.send(endpoint, {
				method: 'GET',
				headers: {
					Accept: 'application/pdf'
					// Authorization: pb.authStore.token
				}
			});
```

What are the limitations of pb.send from normal `fetch`?

**Comment by ganigeorgiev** [comment]
I don't understand what you mean by "download the file" (or what you are trying to do to be honest).

The first example doesn't do that. The raw `fetch` returns a `Response` object and you need to call `json()`/`bytes()`/`text()`/`blob()`/etc. on it to convert the body content.

The JS SDK `send` expects at the moment the response to be json (see [Client.ts#L353](https://github.com/pocketbase/js-sdk/blob/01951f3a42152e72b79ed5709435edf94dca279f/src/Client.ts#L353)). I'll consider in the future to add support for other data types based maybe on the response content-type header but for now remains a low priority.

**Reply by mukhtharcm** [reply]
Oh!

I was trying to send a file from an endpoint from pocketbase.
since you clarified that the send expects a `json` response, this is out of the league I suppose.

Thank you for the response 🙂
